### ISSUE TITLE
Trying to print a bad tank crashes the urbit

### ISSUE THREAD

**Issue by Fang-** [urbit/urbit#4169]
Trying to evaluate the following expression in dojo (or including it in an app, lib, or elsewhere) crashes the urbit process:

```hoon
|%  ++  x  1  ++  x  2  --
```

```
fish: 'pkg/urbit/build/urbit nec' terminated by signal SIGABRT (Abort)
newt: write failed broken pipe
serf: pier error: broken pipe
```

I vaguely remember this resulting in a legible complaint before. Certainly haven't seen it crash like this before, but maybe I just never hit it?

**Comment by joemfb** [comment]
It's crashing in the king due to a bad `$tank (which we `+wash` without virtualization).

**Comment by joemfb** [comment]
Attaching a debugger, the bad tank is:

```
> =a `(list @)`[100 117 112 108 105 99 97 116 101 32 97 114 109 58 32 43 120 ~]
> ;;(tape a)
"duplicate arm: +x"
```

In `+open:ap`:

```
[%eror *]  ~>(%slog.[0 leaf=p.gen] !!)
```

**Comment by philipcmonk** [comment]
That should be `leaf+p.gen`, right?  Or else just a cord I suppose

**Comment by Fang-** [comment]
Fixed as of #4176.
